container/dockershell.sh

```diff
@@ -49,39 +84,74 @@ while [ -n "$1" ]; do
         shift 2
         ;;
     *)
-        echo "Unknown option:$1"
+        printf "Unknown option:$1"
         usage >&2
         exit 1
     esac
 done
 
 if [ -z ${EVENT_USER} ]; then
-    echo "Error: Please provide the Watson Studio Local user name with --user flag"
+    printf "Error: Please provide the cluster user name with --user flag\n"
     usage >&2
     exit 1
 fi
 
 if [ -z ${EVENT_PASSWORD} ]; then
-    echo "Error: Please provide the Watson Studio Local password with --password flag"
+    printf "Error: Please provide the cluster password with --password flag\n"
     usage >&2
     exit 1
 fi
 
 if [ -z ${IP} ]; then
-    echo "Error: Please provide the Event Store server's public IP with --IP flag"
+    printf "Error: Please provide the Event Store server's public IP with --IP flag\n"
     usage >&2
     exit 1
 fi
 
+if [ -z ${DTYPE} ]; then
+    printf "DeploymentType (--deploymentType) not specified, defaulting too: $defaultDeployType\n"
+    DTYPE=$defaultDeployType
+fi
+
+case "$DTYPE" in
+   $deployTypeCp4d)
+      if [ -z ${SERVICE_NAME} ]; then
+         printf "Error: cp4d deployment types require a service name (i.e. --service_name) as input\n"
+         usage >&2
+         exit 1
+      fi
+      ;;
+   $deployTypeDsx)
+      ;;
+   $deployTypeStandalone)
+      ;;
+   *)
+      printf "Deployment type \"$DTYPE\" not supported!\n"
+      usage >&2
+      exit 1
+esac
+
+if [ -z ${IPREST} ]; then
+   printf "Rest Deployment IP (--IPR) not specified, using the (--IP) endpoint: $IP\n"
+   IPR=$IP
+fi
+
 mkdir -p ${USER_VOLUME}
 # start container in interactive mode
 
+entryPoint="env && ${SETUP_PATH}/setup-ssl.sh && ${SETUP_PATH}/entrypoint_msg.sh && bash --login"
+
+# For standalone deployment types there SSL is not enabled so do not execute the
+# corresponding setup.
+if [ $DTYPE = $deployTypeStandalone ]; then
+   entryPoint="${SETUP_PATH}/entrypoint_msg.sh && bash --login"
+fi
+
 docker run -it --name eventstore_demo_${EVENT_USER} -v ${USER_VOLUME}:/root/user_volume \
-    -e EVENT_USER=${EVENT_USER} -e EVENT_PASSWORD=${EVENT_PASSWORD} -e IP=${IP} \
-    eventstore_demo:latest \
-    bash -c "${SETUP_PATH}/setup-ssl.sh && ${SETUP_PATH}/entrypoint_msg.sh && bash --login"
+    -e EVENT_USER=${EVENT_USER} -e EVENT_PASSWORD=${EVENT_PASSWORD} -e IP=${IP} -e IPREST=${IPREST} -e DTYPE=${DTYPE} -e SERVICE_NAME=${SERVICE_NAME}\
+    eventstore_demo:latest bash -c "$entryPoint"
 
-echo "Cleaning up dangling images and/or exited containers"
+printf "Cleaning up dangling images and/or exited containers"
 docker rmi $(docker images -q -f dangling=true) > /dev/null 2>&1
```

Comment:
is this guaranteed not to fail ? (ie not require -f ?)

Reply:
No it's needed, I think we should revisit this as we blindly delete any stopped containers the line below, ideally we'd just work on the container we started and our images.  Will revisit down the road.